tests/test_downloader_handlers.py

```diff
@@ -437,6 +437,8 @@ class S3AnonTestCase(unittest.TestCase):
         import boto
     except ImportError:
         skip = 'missing boto library'
+    if six.PY3:
+        skip = 'S3 not supported on Py3'
```

Comment:
Missed that - this is good! Then s3 support can be added in a separate PR (did a quick check that it is not completely trivial).